x-pack/plugins/security_solution/public/management/components/artifact_list_page/hooks/use_artifact_delete_item.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { useMutation, UseMutationResult } from 'react-query';
+import { i18n } from '@kbn/i18n';
+import { useMemo } from 'react';
+import { useToasts } from '../../../../common/lib/kibana';
+import { ExceptionsListApiClient } from '../../../services/exceptions_list/exceptions_list_api_client';
+
+export const ARTIFACT_DELETE_ACTION_LABELS = Object.freeze({
+  deleteActionFailure: (itemName: string, errorMessage: string) =>
+    i18n.translate('xpack.securitySolution.deleteActionFailure', {
+      defaultMessage: 'Unable to remove "{itemName}" . Reason: {errorMessage}',
+      values: { itemName, errorMessage },
+    }),
+
+  deleteActionSuccess: (itemName: string) =>
+    i18n.translate('xpack.securitySolution.hostIsolationExceptions.deletionDialog.deleteSuccess', {
+      defaultMessage: '"{itemName}" has been removed',
+      values: { itemName },
+    }),
+});
+
+type UseArtifactDeleteItemMutationResult = UseMutationResult<
+  ExceptionListItemSchema,
+  Error,
+  ExceptionListItemSchema
+>;
+
+export type UseArtifactDeleteItemInterface = UseArtifactDeleteItemMutationResult & {
+  deleteArtifactItem: UseArtifactDeleteItemMutationResult['mutateAsync'];
+};
+
+export const useArtifactDeleteItem = (
+  apiClient: ExceptionsListApiClient,
+  labels: typeof ARTIFACT_DELETE_ACTION_LABELS
+): UseArtifactDeleteItemInterface => {
+  const toasts = useToasts();
+
+  const mutation = useMutation(
+    async (item: ExceptionListItemSchema) => {
+      return apiClient.delete(item.id);
+    },
+    {
+      onError: (error: Error, item) => {
+        toasts.addDanger(labels.deleteActionFailure(item.name, error.message));
+      },
+      onSuccess: (response) => {
+        toasts.addSuccess(labels.deleteActionSuccess(response.name));
+      },
+    }
+  );
+
+  return useMemo(() => {
+    return {
+      ...mutation,
+      deleteArtifactItem: mutation.mutateAsync,
```

Comment:
I'm wondering why do you need the mutateAsync? If this is going to be used in a component, you can use the `mutate` function that will trigger the action and will render the component depending on the updated props (isLoading, data, etc). The mutate async requires an await and it's something you cannot do in whatever place when in a component context. If this is not needed, I would suggest just return the `useMutation(...` without adding it in a useMemo.

Reply:
@dasansol92 

You are right in that we don't explicitly need it. I just wanted to have a member in the object whose name was associated with the hook's purpose. `.mutate()` is kind of a weird name to me when working within a specific context and I bet that be renaming that in the components anyway in order to be clearer espcially as we use other actions like edit, etc):

```typescript
const { mutate: deleteArtifactItem } = useArtifactDeleteItem();

const { mutate: editArtifactItem } = useArtifactEditItem();

```

You should also consider this approach in the generic hooks you are creating.

re: `useMemo` - that is actually something I want to test to see if it is even worth it. I need to see how often the `mutation` result changes between re-renders that don't change any data on it

re: use of `mutateAsync`
The reason I picked `mutateAsync` vs `mutate` is because that can same some useEffect() usages or conditions checks in components when trying to determine if a "refresh" should happen. A good example is the code I have in the `ArtifactDeleteModal`, where once the update is successful, call the `onSuccess()` callback (which in this case trigger the list page to refetch the list data) - 

```typescript
const onConfirm = useCallback(() => {
      deleteArtifactItem(item).then(() => onSuccess());
    }, [deleteArtifactItem, item, onSuccess]);
```

Reply:
Just saw [this](https://react-query.tanstack.com/guides/migrating-to-react-query-3#mutationmutate-no-longer-return-a-promise), so ok for using `mutateAsync` 👍